### PR TITLE
Milestone: Capstone project :rocket: :100: 

### PR DESCRIPTION
**Pull request by ahmedeid6842**
## Description: :speech_balloon: 
This pull request for illustrate what I've done to finish the budget mobile app using RoR 

## Done: :white_check_mark: 
- [x] Create a rails project 
- [x] Setup the linters, rubocop & styleint
- [x] Setup the `devise` for authentication  
- [x] Added the navbar
- [x] Setup the home page
- [x] Setup bootstrap    

- [x] Create groups table
- [x] Create deal table
- [x] Adding the references between tables     

- [x] Create the data models for deals and groups, and set up the relationships between them.
- [x] Create the controllers and views for deals and groups, and set up the routes.
- [x] Add the deals and groups index and new views.
- [x] Add the create, index, and show actions to the deals and groups controllers.
- [x] Update the deals and groups tables to add the required columns and foreign keys.
- [x] Add the splash controller and view, and set up the routes.

- [x] Setup and Using Rspec for test 
- [x] Implement different test for different components to ensure it's consistency. 

- [x] Adding README
- [x] Record A demo [video](https://drive.google.com/file/d/1AaFD5X6XaqG90uHn7fFfiAOm69FpB6ge/view?usp=share_link)  

### [Demo Video :eyes: :movie_camera:](https://drive.google.com/file/d/1AaFD5X6XaqG90uHn7fFfiAOm69FpB6ge/view?usp=share_link)

--- 
<p align="center"> @ahmedeid6842 </p>
<p align="center">
  200w
</p>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,199 @@
...
+- [📖 About the Project](#about-project)
+  - [🛠 Built With](#built-with)
+    - [Tech Stack](#tech-stack)
+- [💻 Getting Started](#getting-started)
+  - [Setup](#setup)
+  - [Prerequisites](#prerequisites)
+  - [Install](#install)
+  - [Usage](#usage)
+  - [Run tests](#run-tests)
+  - [Deployment](#triangular_flag_on_post-deployment)
```
Comment (TedLivist) [inline README.md:21]: - [ ] These two links are broken. Kindly fix them or remove them.

- [ ] Kindly deploy your app and add the link in the README

- [ ] Your video demo length is longer than 5 minutes. Kindly provide video presentation that is less than 5 minutes in length based on the requirements.

File: app/views/deals/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  
+      <div class="field my-3">
+        <%= f.select :groups, options_from_collection_for_select(@groups, 'id', 'name'), {}, { class: "form-control" }%>
+      </div>
+  
+      <div class="text-center">
+        <%= f.submit "Add Deal",  class: "btn btn-primary" %>
+      </div>
+    <% end %>
+  </section>
```
Comment (TedLivist) [inline app/views/deals/new.html.erb:19]: - [ ] From the requirement I am on the new transaction page, there should be a "<" button on the navbar that takes me to the transactions index page when I click it, but you do not have the button. Kindly add this arrow button to the navbar and ensure it takes users to the transaction index page.

  image

File: app/views/deals/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        </div>
+    <% end %>
+  
+    <div class="fixed-bottom mb-3 text-center">
+     <%= link_to "New Deal", new_group_deal_path, class:"btn btn-success fs-3 w-75"  %>
+    </div>
+  
+  </section>
+  
+
```
Comment (TedLivist) [inline app/views/deals/index.html.erb:22]: - [ ] When I am on the transaction list page, there should be an arrow button ("<") that takes me to the home page. Kindly add this button and ensure it works as expected.

  image

File: app/views/groups/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        <%= form.text_field :icon, class: "form-control w-100", placeholder: "Icon" %>
+      </div>
+  
+      <div class="text-center m-3">
+        <%= form.submit "Save", class: "btn btn-primary" %>
+      </div>
+  
+    <% end %>
+  
+  </section>
```
Comment (TedLivist) [inline app/views/groups/new.html.erb:18]: - [ ] There should be an arrow back button on the new category page that takes users to the home page. Kindly add it.

  image

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @ahmedeid6842,

Good job so far!
✔️ Implementation of splash screen
✔️ Well-written and passing tests
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (ahmedeid6842) [reply README.md]: Fixed

File: app/views/deals/new.html.erb (hunk 2 above)
Comment (ahmedeid6842) [reply app/views/deals/new.html.erb:19]: Fixed

File: app/views/deals/index.html.erb (hunk 3 above)
Comment (ahmedeid6842) [reply app/views/deals/index.html.erb:22]: Fixed

File: app/views/groups/new.html.erb (hunk 4 above)
Comment (ahmedeid6842) [reply app/views/groups/new.html.erb:18]: Fixed

**Comment by ahmedeid6842** [comment]
### According to last review and required changes I recorded a new video with < 5 minutes
### [Demo Video :eyes: :movie_camera:](https://drive.google.com/file/d/1AaFD5X6XaqG90uHn7fFfiAOm69FpB6ge/view?usp=share_link)

### Also I deployed [Live Demo](https://budget-app-microverse-79ca66bb0c60.herokuapp.com/) to Heroku. 
### I added too the different arrow backs (<-) based on the review to move between different pages

File: db/schema.rb [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+#
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema[7.0].define(version: 2023_10_19_024538) do
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
+
+  create_table "deals", force: :cascade do |t|
+    t.string "name", null: false
+    t.integer "author_id", null: false
```
Comment (FrederickMih) [inline db/schema.rb:19]: - [Optional] I think integer datatype is not big enough for an id data. Most often bigint is recommended :+1:

File: spec/views/signup_spec.rb [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  end
+
+  it "should error appears if the password don't match confirm" do
+    fill_in 'Name', with: @user1.name
+    fill_in 'Email', with: @user1.email
+    fill_in 'New password', with: @user1.password
+    fill_in 'Confirm Password', with: ''
+    click_button 'Sign up'
+    expect(page).to have_content('1 error')
+    expect(page).to have_content('Password confirmation doesn\'t match Password')
```
Comment (FrederickMih) [inline spec/views/signup_spec.rb:49]: - [Optional] Please kindly fix the failing tests as you can see below:
Screenshot from 2023-10-20 06-16-00

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand: :raising_hand: @ahmedeid6842  , 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights :green_circle: 

- Good job for adding only necessary routes for your controller's actions :+1: 

## Optional suggestions

- It is always recommendation to follow the correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) for any solo project you are building. This project will be deploy directly to the main branch which is not accepted. You may commit some errors and your app will still be deploy automatically. Always create features branch and make the dev branch default for automatic deployment. Open PR from feature to dev branch

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by ahmedeid6842** [comment]
Thanks @FrederickMih for your time and effort to review my code. :pray: :heart_decoration:

**Pull request merged**